src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -1,47 +1,151 @@
+
+    List<SelectExpression> select = List.of(nameFieldVariable, idFieldVariable, countFieldVariable);
+
+    List<TableVariable> tables =
+        List.of(conceptTableVariable, conceptAncestorTableVariable, domainOccurenceTableVariable);
 
-    TablePointer ancestorTablePointer =
-        TablePointer.fromTableName("concept_ancestor", tableNameGenerator);
-    TableVariable ancestorTableVariable = TableVariable.forPrimary(ancestorTablePointer);
-    FieldVariable descendantFieldVariable =
-        ancestorTableVariable.makeFieldVariable("descendant_concept_id");
+    List<FieldVariable> groupBy = List.of(nameFieldVariable, idFieldVariable);
 
-    BinaryFilterVariable whereClause =
+    List<OrderByVariable> orderBy =
+        List.of(new OrderByVariable(nameFieldVariable, OrderByDirection.ASCENDING));
+
+    // ancestorTable is primary table for the subquery
+    var ancestorTablePointer = TablePointer.fromTableName(CONCEPT_ANCESTOR, tableNameGenerator);
+    var ancestorTableVariable = TableVariable.forPrimary(ancestorTablePointer);
+    var descendantFieldVariable = ancestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID);
+
+    // WHERE c.ancestor_concept_id = conceptId
+    BinaryFilterVariable ancestorClause =
         new BinaryFilterVariable(
-            ancestorTableVariable.makeFieldVariable("ancestor_concept_id"),
+            ancestorTableVariable.makeFieldVariable(ANCESTOR_CONCEPT_ID),
             BinaryFilterVariable.BinaryOperator.EQUALS,
             new Literal(conceptId));
+
+    // WHERE d.descendant_concept_id != conceptId
+    BinaryFilterVariable notSelfClause =
+        new BinaryFilterVariable(
+            ancestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID),
+            BinaryFilterVariable.BinaryOperator.NOT_EQUALS,
+            new Literal(conceptId));
+
+    // WHERE c.ancestor_concept_id = conceptId AND d.descendant_concept_id != conceptId
+    List<FilterVariable> clauses = List.of(ancestorClause, notSelfClause);
+    BooleanAndOrFilterVariable subqueryWhereClause =
+        new BooleanAndOrFilterVariable(BooleanAndOrFilterVariable.LogicalOperator.AND, clauses);
+
+    // (SELECT c.descendant_concept_id FROM concept_ancestor AS c
+    // WHERE c.ancestor_concept_id = conceptId)
     Query subQuery =
-        new Query(List.of(descendantFieldVariable), List.of(ancestorTableVariable), whereClause);
-    SubQueryFilterVariable subQueryFilterVariable =
-        new SubQueryFilterVariable(idFieldVariable, SubQueryFilterVariable.Operator.IN, subQuery);
-    Query query =
         new Query(
-            List.of(nameFieldVariable, idFieldVariable),
-            List.of(conceptTableVariable),
-            subQueryFilterVariable);
+            List.of(descendantFieldVariable), List.of(ancestorTableVariable), subqueryWhereClause);
+
+    // WHERE c.concept_id IN subQuery
+    SubQueryFilterVariable where =
+        new SubQueryFilterVariable(idFieldVariable, SubQueryFilterVariable.Operator.IN, subQuery);
```

Comment:
And a helper for `in`
```suggestion
    SubQueryFilterVariable where = SubQueryFilterVariable.in(idFieldVariable, subQuery);
```